src/components/Categories/Categories.js

```diff
@@ -0,0 +1,78 @@
+import {useEffect, useState} from 'react';
+import styled from 'styled-components';
+
+export default function Categories({menus, handleCategoryClick, curClickedCategory}) {
+  const [categories, setCategories] = useState([]);
+
+  useEffect(() => {
+    /* Get categories form the given menus */
+    function getCategories(menus) {
```

Comment:
You should test such complex functions.

Reply:
But you dont test such functions with junit tests.

Reply:
> 
Ok, I see. Students from other group make junit tests? In my group, Andrea said, it's not necessary. Component test is a must, I can try to test this function in component tests though.

Reply:
Depends on the complexity.